miner/miner.go

```diff
@@ -67,132 +64,145 @@ type Miner struct {
 	txFeeRecipient common.Address
```

Comment:
txFee can go. We definitely still need `validator` since not being able to `recoverRandomness` causes a lot of problems, (it also takes a while, so it's important that if it needs to happen, it gets done in a one-shot event).